elastic/fleet-server
# Prometheus metrics

## What is the problem this PR solves?

Expose fleet-server metrics in prometheus format. Ship metrics collected by prometheus with APM tracer.

## How does this PR solve the problem?

Wrap custom metrics collection so we still collect the legacy libbeat metrics along the new prometheus instrumentation.
Ship prometheus instrumented metrics with the `apm.Tracer`.

## How to test this PR locally

Start server with `http.enabled: true` and check the `/metrics` endpoint of that listener (on `http://localhost:5066` by default).

Or setup APM tracing and view metrics.

## Design Checklist

N/A

## Checklist

- ~~I have made corresponding changes to the documentation~~
- ~~I have made corresponding change to the default configuration files~~

## Related issues

- Closes #2542
